src/controllers/QueryController.ts

```diff
@@ -115,15 +116,21 @@ export class QueryController extends RootComponent {
    * Create a new query controller
    * @param element
    * @param options
+   * @param usageAnalytics @deprecated Will use the {@link IAnalyticsClient} from the @param searchInterface instead.
```

Comment:
This gets rendered as such:

![Screen Shot 2019-10-02 at 9 00 08 AM](https://user-images.githubusercontent.com/23503066/66046209-4a861d80-e4f3-11e9-923f-7bb56f8c672e.png)

Perhaps we should write instead:

```
**Deprecated.** Since the [October 2019 Release (v2.7219)](https://docs.coveo.com/en/3084/), the class retrieves and uses the {@link AnalyticsClient} from its `searchInterface` constructor parameter.
```

I wonder, though... since the parameter is still required, should the public documentation even indicate that it's deprecated? Should we rather leave a comment in the code to explain that the class no longer uses it, but it has to remain there to avoid introducing a potentially breaking change?

Reply:
We could, but it's also important to they the people using that class that it's going to "retrieves and uses the {@link AnalyticsClient} from its `searchInterface` constructor parameter"

Knowing that, should we change the deprecated doc?

Reply:
The only thing that slightly bothers me is that the parameter is deprecated _and_ required...

How about:

```
**Deprecated.** Still required, but no longer used since the [October 2019 Release (v2.7219)](https://docs.coveo.com/en/3084/). From this version on, the class retrieves and uses the {@link AnalyticsClient} from its `searchInterface` constructor parameter instead.
```
?